flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractFetcher.java

```diff
@@ -435,6 +435,13 @@ private List<KafkaTopicPartitionState<T, KPH>> createPartitionStateHolders(
                                         kafkaHandle,
                                         deserializedWatermarkStrategy.createTimestampAssigner(
                                                 () -> consumerMetricGroup),
+                                        // When upgrading to Flink 2.0, context has to provide also
+                                        // the input activity clock. This is not trivial for the old
+                                        // sources. Ideally we should drop this old source before
+                                        // this connector is upgraded to Flink 2.0. Otherwise, we
+                                        // can avoid the compilation error without fixing the
+                                        // FLIP-471 bug, by returning SystemClock as the input
+                                        // activity clock. Note that would be incorrect thing to do.
```

Comment:
```suggestion
                                        // activity clock. Note that would be incorrect thing to do.
```

Wouldn't that more or less just reproduce the status quo? So while incorrect, it wouldn't be a regression.

If so, I'd rephrase `by returning SystemClock, which would reproduce the pre-FLIP-471 behavior`.